### PR TITLE
linked time: set up feature flag and fob rendering

### PR DESCRIPTION
**Pull request by japie1235813**
* Motivation for features / changes
Introduce feature flag in the card fob and used to render a fob component.

* Technical description of changes
Getting prospectiveFob feature and passing all the way down to card-fob.

Note: this part might need refactor once we have nested container or some kind of structure refactoring.

* Screenshots of UI changes
None

### CODE REVIEW

File: tensorboard/webapp/widgets/card_fob/card_fob_controller_test.ts [hunk 1]
```diff
@@ -85,6 +89,8 @@ describe('card_fob_controller', () => {
     timeSelection: TimeSelection;
     showExtendedLine?: Boolean;
     steps?: number[];
+    isProspectiveFobFeatureEnabled?: Boolean;
+    prospectiveStep?: number | null;
```
Comment (rileyajones) [inline tensorboard/webapp/widgets/card_fob/card_fob_controller_test.ts:93]: nitpick: Why make this null rather than `undefined`?
```suggestion
    prospectiveStep?: number;
```

File: tensorboard/webapp/widgets/card_fob/card_fob_controller_component.ts [hunk 2]
```diff
@@ -58,6 +60,8 @@ export class CardFobControllerComponent {
   @Input() highestStep!: number;
   @Input() lowestStep!: number;
   @Input() showExtendedLine?: Boolean = false;
+  @Input() isProspectiveFobFeatureEnabled: Boolean = false;
+  @Input() prospectiveStep: number | null = null;
```
Comment (rileyajones) [inline tensorboard/webapp/widgets/card_fob/card_fob_controller_component.ts:64]: nitpick: Why make this null rather than `undefined`?
```suggestion
  @Input() prospectiveStep?: number;
```

**Review by rileyajones — APPROVED** [review]

File: tensorboard/webapp/widgets/card_fob/card_fob_controller_component.ts (hunk 2 above)
Comment (japie1235813) [reply tensorboard/webapp/widgets/card_fob/card_fob_controller_component.ts:64]: prospectiveStep will be set to null on mouseleave event. While undefined means the variable is not defined, assigning it to null makes more sense to me.
Is this reasonable to you? Or any other strong reason to use undefined?

I noticed this two inputs should be optional. Adding question marks.

File: tensorboard/webapp/widgets/card_fob/card_fob_controller_test.ts (hunk 1 above)
Comment (japie1235813) [reply tensorboard/webapp/widgets/card_fob/card_fob_controller_test.ts:93]: Different from the code, the test here makes sense to just use undefined, since it will be assigned null in line 144.

Comment (japie1235813) [reply tensorboard/webapp/widgets/card_fob/card_fob_controller_test.ts:93]: Ah my bad, I still need null here since I want to test element rendeing on step being null
